fix jit trace error for model forward sequence is not aligned with jit.trace tuple input sequence, update related doc (#19891)
huggingface/transformers: docs/source/en/perf_infer_cpu.mdx

IPEX release is following PyTorch, check the approaches for [IPEX installation](https://intel.github.io/intel-extension-for-pytorch/).

### Usage of JIT-mode
To enable JIT-mode in Trainer for evaluaion or prediction, users should add `jit_mode_eval` in Trainer command arguments.

<Tip warning={true}>

for PyTorch >= 1.14.0. JIT-mode could benefit any models for prediction and evaluaion since dict input is supported in jit.trace

> for PyTorch < 1.14.0. JIT-mode could benefit models like "Question and Answer", whose forward parameter order matches the tuple input order in jit.trace.

Reviewer: ```suggestion
for PyTorch < 1.14.0. JIT-mode could benefit modelswhose forward parameter order matches the tuple input order in jit.trace, like question-answering models.
```

> If we meet case like "text classification", whose forward parameter order does not match the tuple input order in jit.trace, jit.trace will fail and we are capturing this with the exception here to make it fallback and logging is used to notify users.

Author: done
